draft-ietf-quic-http.md

```diff
@@ -1099,36 +1104,39 @@ the RST bit set if it detects an error with the stream or the QUIC connection.
-upon another element. This information is expressed in the PRIORITY frame
-{{frame-priority}} which identifies the element and the dependency. The elements
-that can be prioritized are:
-
-- Requests, identified by the ID of the request stream
+5.3, but replaces streams depending upon streams with strict priorities and
+placeholders.  In the HTTP/3 priority scheme, a stream can be given a strict
+priority with weights for equal priority elements, or can be designated as
+dependent upon a placeholder or the root.  This information is expressed in the
+PRIORITY frame {{frame-priority}} which identifies the element and the
+dependency. The elements that can be prioritized are:
+
+- Requests, identified by the stream of the PRIORITY frame
 - Pushes, identified by the Push ID of the promised resource
   ({{frame-push-promise}})
 - Placeholders, identified by a Placeholder ID
 
-Taken together, the dependencies across all prioritized elements in a connection
-form a dependency tree. An element can depend on another element or on the root
-of the tree. A reference to an element which is no longer in the tree is treated
-as a reference to the root of the tree. The structure of the dependency tree
-changes as PRIORITY frames modify the dependency links between prioritized
-elements.
-
-Due to reordering between streams, an element can also be prioritized which is
-not yet in the tree. Such elements are added to the tree with the requested
-priority.
+In HTTP/3, stream dependencies that were implicitly encoded by dependencies in
+HTTP/2 are explicitly encoded with strict prioritization.
 
 When a prioritized element is first created, it has a default initial weight
-of 16 and a default dependency. Requests and placeholders are dependent on the
-root of the priority tree; pushes are dependent on the client request on which
-the PUSH_PROMISE frame was sent.
+of 0, priority of 16, and a default dependency. Requests and placeholders are
+dependent on the root of the priority tree; pushes are dependent on the same
+parent as the client request on which the PUSH_PROMISE frame was sent and given
+a priority value one smaller than the request stream.
 
 Requests may override the default initial values by including a PRIORITY frame
 (see {{frame-priority}}) at the beginning of the stream. These priorities
 can be updated by sending a PRIORITY frame on the control stream.
 
+Higher priority elements have all available data sent before elements of lower
+priority.  When multiple elements have the same priority, if a weight is
+specified, elements with weights are interleaved.  If a weight is 0, either
+because it is the default weight or it was not specified, then elements are
+delivered sequentially and all available data from one element is sent before
+any data from the next.  If a given priority level has some elements with
+weights and some without, the two groups share the available bandwidth equally.
```

Comment:
You can have 10 elements with weights at some priority level, sharing bandwidth.  When you add a single element to that priority level without a weight, the bandwidth for the 10 elements is halved.  This does not seem right.

```suggestion
weights and some without, the two groups share the available bandwidth proportionally to the number of elements in each group.
```

Reply:
I suggest looking at rmarx proposal.

Reply:
Good point @igorlord This was reflecting what Patrick Meenan wrote up, but I agree that it's not clear it's the right option.

Personally, I would prefer removing weights from streams and only having one bit to declare that this resource is incrementally useful.